config/components/local-db/kustomization.yaml

```diff
@@ -22,5 +22,9 @@ kind: Component
 resources:
 - 201-sql-deployment.yaml
 
-patchesStrategicMerge:
-- api.yaml
+patches:
+- target:
+    kind: Deployment
+    name: api
+    namespace: tekton-pipelines
```

Comment:
Do we need this, since `api.yaml` contains the target object? Or is this considered best practice?

Reply:
The target element is not optional.

Reply:
`target` field is not mandatory for strategic merge.
If `target` field is omitted, the target resource is matched using the `apiVersion`, `kind` and `name` from the patch.

https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/#patch-using-path-strategic-merge